textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -41,6 +48,11 @@
     private ScoreDoc[] scoreDocs;
     private IndexReader luceneIndexReader;
     private DataReaderPredicate dataReaderPredicate;
+    private ArrayList<String> queryTokens;

```

Comment:
Interesting thought.  I wonder if a variable is only used in getNextTuple() as a local variable (allocated on the stack), does it help by declaring it globally within the class.  From the readability perspective, it's better to declare it locally.